### ISSUE TITLE
Lutris cannot run Windows executables with Wine due to rfind error

### ISSUE THREAD

**Issue by CartoonFan** [lutris/lutris#2369]
**Describe the bug**

Trying to run any Windows executable shows "AttributeError: 'NoneType' object has no attribute 'rfind". If using the play button, the terminal also shows that the game prelaunch was unsuccessful.

**Expected behavior**

Lutris should be able to run Windows executables using Wine.

**Current behavior**

Using Wine to run Windows executables in Lutris is not currently possible for me.

**Steps to reproduce**

- Press Play button on a Wine game in Lutris.

**Lutris debugging output**

INFO     2019-09-28 09:55:16,955 [application.do_command_line:225]:Running Lutris 0.5.3
INFO     2019-09-28 09:55:16,955 [startup.check_driver:54]:Using NVIDIA drivers 435.24.02 for x86_64
INFO     2019-09-28 09:55:16,955 [startup.check_driver:58]:GPU: GeForce GTX 960
INFO     2019-09-28 09:55:16,956 [startup.check_driver:75]:GPU: 8086:0402 1043:8534 using i915 drivers
INFO     2019-09-28 09:55:16,988 [startup.check_vulkan:129]:Vulkan is supported
INFO     2019-09-28 09:55:16,989 [dxvk.get_dxvk_versions:21]:Updating DXVK versions
INFO     2019-09-28 09:55:17,381 [dxvk.get_dxvk_versions:21]:Updating D9VK versions
DEBUG    2019-09-28 09:55:17,515 [http.get:65]:GET https://lutris.net/api/runtime
INFO     2019-09-28 09:55:17,522 [lutriswindow.toggle_connection:580]:Connected to lutris.net as CartoonFan
DEBUG    2019-09-28 09:55:17,528 [http.get:65]:GET https://lutris.net/api/games/library/CartoonFan
DEBUG    2019-09-28 09:55:17,538 [store.get_missing_media:194]:Requesting missing icons from API for 131 games
DEBUG    2019-09-28 09:55:17,539 [http.get:65]:GET https://lutris.net/api/games
DEBUG    2019-09-28 09:55:19,091 [api.get_game_api_page:132]:Loaded 130 games from page 1
DEBUG    2019-09-28 09:55:25,124 [application.launch:339]:Launching Ys Origin (wine) (140341854858608)
DEBUG    2019-09-28 09:55:25,172 [game.play:278]:Prelaunching <lutris.runners.wine.wine object at 0x7fa3e3a6b750>
ERROR    2019-09-28 09:55:25,559 [jobs.target:32]:Error while completing task <bound method wine.prelaunch of <lutris.runners.wine.wine object at 0x7fa3e3a6b750>>: 'NoneType' object has no attribute 'rfind'
<class 'AttributeError'> 'NoneType' object has no attribute 'rfind'
  File "/usr/lib/python3.7/site-packages/lutris/util/jobs.py", line 30, in target
    result = self.function(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/lutris/runners/wine.py", line 776, in prelaunch
    self.sandbox(prefix_manager)
  File "/usr/lib/python3.7/site-packages/lutris/runners/wine.py", line 927, in sandbox
    wine_prefix.desktop_integration(restore=True)
  File "/usr/lib/python3.7/site-packages/lutris/util/wine/prefix.py", line 86, in desktop_integration
    DESKTOP_FOLDERS.append(folder[folder.rfind("\\")+1:])
ERROR    2019-09-28 09:55:25,560 [game.configure_game:295]:'NoneType' object has no attribute 'rfind'
ERROR    2019-09-28 09:55:26,696 [game.configure_game:298]:Game prelaunch unsuccessful

**System information**

[lutris-issue-report_2019_09_28.json.zip](https://github.com/lutris/lutris/files/3665765/lutris-issue-report_2019_09_28.json.zip)

**Screenshots**

![20190928-100025](https://user-images.githubusercontent.com/29664894/65819896-e1429980-e1d6-11e9-9713-59de3243aa20.png)
![20190928-100027](https://user-images.githubusercontent.com/29664894/65819897-e1429980-e1d6-11e9-8f7c-168421acf8c2.png)

**Comment by G4E** [comment]
Hello, what is your locale ? It seems that it is en_US from the report you shared.
From what I see you disabled the wine's sandbox. You're trying to launch Ys Origin, correct ?
Could you do 
```
ls -l ~/Games/[ys_origin_wine_prefix]/drive_c/users/jeremiah/ 
``` 
And could you share the file ~/Games/[ys_origin_wine_prefix]/user.reg ?
Thank you !

**Comment by CartoonFan** [comment]
My locale is en_US.UTF-8. I am trying to use Ys Origin, but I remember being able to reproduce this with other games as well. I've been using a single prefix for pretty much everything instead of a prefix-per-game, but here you go:

```
ls -l /home/jeremiah/games/lutris/wine_dxvk_and_d9vk/drive_c/users/jeremiah

total 572
drwxr-xr-x 1 jeremiah users     16 Aug 22 23:25  AppData
drwxr-xr-x 1 jeremiah users     30 Sep 14 01:00 'Application Data'
drwxr-xr-x 1 jeremiah users      0 Aug 22 23:25  Contacts
drwxr-xr-x 1 jeremiah users     18 Aug 24 02:28  Cookies
lrwxrwxrwx 1 jeremiah users     15 Oct  5 01:41  Desktop -> /home/jeremiah/
drwxr-xr-x 1 jeremiah users      6 Sep 14 01:15  Documents
drwxr-xr-x 1 jeremiah users      0 Aug 22 23:25  Downloads
drwxr-xr-x 1 jeremiah users      0 Aug 22 23:25  Favorites
drwxr-xr-x 1 jeremiah users      0 Aug 22 23:25  Links
drwxr-xr-x 1 jeremiah users     94 Aug 22 23:25 'Local Settings'
lrwxrwxrwx 1 jeremiah users     15 Oct  5 01:41 'My Documents' -> /home/jeremiah/
lrwxrwxrwx 1 jeremiah users     15 Oct  5 01:41 'My Music' -> /home/jeremiah/
lrwxrwxrwx 1 jeremiah users     15 Oct  5 01:41 'My Pictures' -> /home/jeremiah/
lrwxrwxrwx 1 jeremiah users     15 Oct  5 01:41 'My Videos' -> /home/jeremiah/
drwxr-xr-x 1 jeremiah users      0 Aug 22 23:25  NetHood
drwxr-xr-x 1 jeremiah users      0 Aug 22 23:25  PrintHood
drwxr-xr-x 1 jeremiah users      0 Aug 22 23:25  Recent
drwxr-xr-x 1 jeremiah users     12 Aug 26 17:31 'Saved Games'
drwxr-xr-x 1 jeremiah users      0 Aug 22 23:25  Searches
drwxr-xr-x 1 jeremiah users      0 Aug 22 23:25  SendTo
drwxr-xr-x 1 jeremiah users     16 Aug 22 23:25 'Start Menu'
drwxr-xr-x 1 jeremiah users   3300 Oct  5 23:40  Temp
drwxr-xr-x 1 jeremiah users      0 Aug 22 23:25  Templates
-rw-r--r-- 1 jeremiah users 134080 Oct  5 22:24  dd_depcheck_NETFX_EXP_35.txt
-rw-r--r-- 1 jeremiah users   2410 Oct  5 22:25  dd_dotnetfx35error.txt
-rw-r--r-- 1 jeremiah users 231854 Oct  5 22:25  dd_dotnetfx35install.txt
-rw-r--r-- 1 jeremiah users  46238 Aug 26 19:05  dd_vcredistMSI4D85.txt
-rw-r--r-- 1 jeremiah users  41842 Aug 26 19:05  dd_vcredistMSI4DF7.txt
-rw-r--r-- 1 jeremiah users  50964 Aug 26 19:17  dd_vcredistMSI5723.txt
-rw-r--r-- 1 jeremiah users  11370 Aug 26 19:05  dd_vcredistUI4D85.txt
-rw-r--r-- 1 jeremiah users  11402 Aug 26 19:05  dd_vcredistUI4DF7.txt
-rw-r--r-- 1 jeremiah users  11666 Aug 26 19:17  dd_vcredistUI5723.txt
-rw-r--r-- 1 jeremiah users   7138 Oct  5 22:25  uxeventlog.txt
```
And here's the zipped user.reg:

[user.reg.zip](https://github.com/lutris/lutris/files/3694566/user.reg.zip)

**Comment by CartoonFan** [comment]
Hmm. I tested it just a little bit after my last post, and it seems to work fine now. I didn't update lutris to the latest master until afterwards, either, so maybe it's a Wine bug? Whatever happened, it seems like it's fixed. I can close the bug now, even though I'm still a little curious what happened.

**Comment by G4E** [comment]
It is definitely not a Wine bug. It is link to a feature which is not yet released (only in lutris git atm). 
What is you distribution ? Your XDG paths are not corrects which is surprising ? 
It look like you tried to launch a game with the wine prefix not yet initialized. I didn't know it was possible within lutris, but as you said you're not following the usual flow of "one prefix for one game".

**Comment by CartoonFan** [comment]
I'm running ArchLabs, a flavor of Arch Linux. As you say, I am running the git version of lutris. I forgot the exact revision I had before, but the problem was fixed before I updated to the latest master. I updated Wine-TKG (also with git) sometime before I tested the issue earlier (maybe yesterday or the day before), so I attributed the change to WIne. I think something weird did happen to my prefix at some point--I had to recently re-add my dll overrides after finding them missing. However, the last time I remember making a new prefix was a while back, and everything seemed fine afterwards. Around the time I created this bug, I changed my Wine version to test a completely different bug, but that didn't cause a problem before. You said that my XDG paths are not correct, but what does that mean?
